provider/provider_nodejs_test.go

```diff
@@ -6,63 +6,60 @@
 package provider
 
 import (
-	"fmt"
+	"os"
 	"path/filepath"
 	"testing"
 
-	"github.com/pulumi/providertest"
-	"github.com/stretchr/testify/assert"
+	"github.com/pulumi/providertest/pulumitest"
+	"github.com/pulumi/providertest/pulumitest/opttest"
+	"github.com/stretchr/testify/require"
 )
 
-func TestLambdaLayerNew(t *testing.T) {
-	nodeTest(t, filepath.Join("..", "examples", "lambda-layer-new"))
+func TestLambdaLayerNewUpgrade(t *testing.T) {
+	testProviderUpgrade(t, filepath.Join("..", "examples", "lambda-layer-new"), nodeProviderUpgradeOpts())
 }
 
-func TestCloudWatch(t *testing.T) {
-	nodeTest(t, filepath.Join("..", "examples", "cloudwatch"))
+func TestCloudWatchUpgrade(t *testing.T) {
+	testProviderUpgrade(t, filepath.Join("..", "examples", "cloudwatch"), nodeProviderUpgradeOpts())
 }
 
-func TestLogGroup(t *testing.T) {
-	nodeTest(t, filepath.Join("..", "examples", "logGroup"))
+func TestLogGroupUpgrade(t *testing.T) {
+	testProviderUpgrade(t, filepath.Join("..", "examples", "logGroup"), nodeProviderUpgradeOpts())
 }
 
-func TestQueue(t *testing.T) {
-	nodeTest(t, filepath.Join("..", "examples", "queue"),
-		providertest.WithSkippedUpgradeTestMode(providertest.UpgradeTestMode_Quick, "Prefer PreviewOnly"),
-		providertest.WithDiffValidation(providertest.NoReplacements()))
```

Comment:
Is the default assert now "no replacements"? Non-blocking but it was previously "no changes" so it'd be nice to bring it back to that, otherwise we'd be losing coverage.

Reply:
Yes. That's intentional. Testing for absence of updates is not a great default for us since it's totally fine with the users in most of the cases but creates friction that makes it difficult to iterate in this repository. I think we need to be looking at stricter checks when it is meaningful to some end user-scenario not here. They would be interesting to have in some kind of statistical/advisory role but the test suite right now is basically pass or fail there is no easy way to track e.g. warnings.